src/internal/DotStarEsp32DmaSpiMethod.h

```diff
@@ -209,10 +224,13 @@ template<typename T_SPISPEED, typename T_SPIBUS> class DotStarEsp32DmaSpiMethod
         {
             _spiTransaction.flags = SPI_TRANS_MODE_QIO;
         }
+        if (T_SPIBUS::ParallelBits == 8)
+        {
+            _spiTransaction.flags = SPI_TRANS_MODE_OCT;
+        }
         _spiTransaction.tx_buffer = _dmadata;
 
         esp_err_t ret = spi_device_queue_trans(_spiHandle, &_spiTransaction, 0);  //Transmit!
-        assert(ret == ESP_OK);            //Should have had no issues.
```

Comment:
Why remove this assert?
Without it you should be getting a warning that ret is not used.  You are compiling with warnings are errors correct?

Reply:
The assert was failing for me, although it showed 0 when I printed it out. So I just removed it - I didn't dig deeper into that.

I'm compiling using platformio with Arduino framework

Reply:
Ok, I just spent some time digging into this. I was wrong. The assert was failing because the queue function was returning ERR_TIMEOUT (I'm not sure why my log didn't show this, oops). I had to fix `IsReadyToUpdate` because it kept returning `true` even though a DMA transfer was still in progress.

Honestly I'm not sure how it ever worked on S2, the logic there is simply wrong. Perhaps things changed between S2 and S3 when the DMA peripheral was upgraded to GDMA. It was "working" on S3 because by removing the assert I was just ignoring those timeout errors and eventually rendering the next frame when DMA became available.

I added the error checking back and tested the fixed code on both S2 and S3.

You are absolutely correct, removing an assert is a terrible idea :)